### PR TITLE
Add University of Wisconsin Madison

### PR DESCRIPTION
**Pull request by aarontorres0**

### CODE REVIEW

File: img/w/wisc.edu.svg [hunk 1]
```diff
@@ -0,0 +1 @@
....45,0.46.38,0.59,0.24,0.59-0.9,0-5.51-.71-5.11-5.22a8,8,0,0,1,3.34-6,7.58,7.58,0,0,1,3.88-1,14,14,0,0,1,2.56.21,8.77,8.77,0,0,1,1,.26q0.52,0.14,1,.32a13.93,13.93,0,0,1,1.93.87,16.19,16.19,0,0,1,4,3.1,20.93,20.93,0,0,1,3.15,4.44A31,31,0,0,1,52.6,22,39.32,39.32,0,0,1,54,27.59c0.35,1.91.58,3.84,0.74,5.77A64.57,64.57,0,0,1,54.7,44.19Z" transform="translate(-0.25 -0.36)"/><path class="cls-3" d="M45,39.63c-0.11-2.69-.9-10.9-6.48-14.46A9.41,9.41,0,0,0,34.18,24c-2.74-.18-4.77.87-6.14,0.91H28c-1.37,0-3.42-1.09-6.16-.91a9.35,9.35,0,0,0-4.37,1.21C11.85,28.73,11.07,36.94,11,39.63a43.52,43.52,0,0,0,3.54,19C18.36,67.12,23,73.14,27.63,75.19L28,75.35l0.37-.16c4.61-2,9.27-8.06,13.14-16.57A43.53,43.53,0,0,0,45,39.63Z" transform="translate(-0.25 -0.36)"/><path class="cls-4" d="M36.3,33.18V35L36.67,35h0.18a1.17,1.17,0,0,1,.82.28,1.27,1.27,0,0,1,.21,1.11s-3.74,16.19-4.45,19.27c-0.82-3.9-5.26-25.18-5.26-25.18l0-.09H27.27v0.1L23.4,55.32,19,36.37a2.7,2.7,0,0,1,0-.28,1.27,1.27,0,0,1,.31-1A1,1,0,0,1,20,34.92l0.37,0v-1.8H14.13v1.76l0.28,0a1.16,1.16,0,0,1,.95.83L23,68.48l0,0.09h1.1v-0.1l3.56-23.3,4.53,23.31,0,0.09h1l7.25-32.78a1,1,0,0,1,1-.75h0.07l0.36,0V33.18H36.3Z" transform="translate(-0.25 -0.36)"/></svg>
```
Comment (Carlgo11) [inline img/w/wisc.edu.svg]:
```suggestion
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 55.5 87.28"><defs><radialGradient id="a" cx=".89" cy="17.74" r="33.44" gradientTransform="matrix(1.09 0 0 1.09 27.04 24.74)" gradientUnits="userSpaceOnUse"><stop offset=".17" stop-color="#fff"/><stop offset=".3" stop-color="#f6ede4"/><stop offset=".69" stop-color="#debe9b"/><stop offset=".87" stop-color="#d4ac7f"/></radialGradient></defs><path fill="url(#a)" d="M28 87.36a3.78 3.78 0 0 1-1.05-1.16l-.06-.09h-.11C8.65 81.86-1.45 54 .83 31.71a47.71 47.71 0 0 1 1.29-7.25 35.39 35.39 0 0 1 2.33-6.39 23.55 23.55 0 0 1 3.75-5.65A14.7 14.7 0 0 1 19 7.28a.33.33 0 0 0 .33-.28 5.49 5.49 0 0 1 3.88-2.6h.19v-.19A4.69 4.69 0 0 1 28 .64a4.76 4.76 0 0 1 4.56 3.54v.19h.19A5.51 5.51 0 0 1 36.67 7a.32.32 0 0 0 .37.26 14.7 14.7 0 0 1 10.77 5.13 24 24 0 0 1 4.24 6.71l.23.55a42.56 42.56 0 0 1 2.89 12C57.45 54 47.35 81.86 29.23 86.08h-.11l-.06.09A3.78 3.78 0 0 1 28 87.36Z" transform="translate(-.25 -.36)"/><path fill="#282728" d="M55.18 31.32a49.49 49.49 0 0 0-.86-5.5 39.81 39.81 0 0 0-1.39-4.93 31.29 31.29 0 0 0-2.23-4.93 22.63 22.63 0 0 0-3-4.1 14.94 14.94 0 0 0-10.95-5.22h-.06a5.77 5.77 0 0 0-4.16-2.91 4.93 4.93 0 0 0-9.56 0 5.77 5.77 0 0 0-4.16 2.91h-.06a14.94 14.94 0 0 0-11 5.23 22.63 22.63 0 0 0-3 4.1 31.29 31.29 0 0 0-2.23 4.93 39.81 39.81 0 0 0-1.39 4.93 49.49 49.49 0 0 0-.86 5.5C-1.98 53.94 8.19 81.74 26.48 86a3.17 3.17 0 0 0 1.27 1.28A3.17 3.17 0 0 0 29.02 86c18.29-4.26 28.46-32.06 26.16-54.68ZM27.75.87a3.9 3.9 0 0 1 3.93 2.77 4.86 4.86 0 0 0-3.86 2.47.17.17 0 0 1-.07.09.15.15 0 0 1-.07-.09 4.86 4.86 0 0 0-3.86-2.47A3.9 3.9 0 0 1 27.75.87Zm-1.86 4.29a3.51 3.51 0 0 1 1.59 2.11.46.46 0 0 0 .27.37.46.46 0 0 0 .27-.42 3.51 3.51 0 0 1 1.59-2.11 4.19 4.19 0 0 1 6 1.58 13.38 13.38 0 0 0-1.67.42 6.6 6.6 0 0 0-2.38 1.32 9.4 9.4 0 0 0-3 6.1c-.67 7.31 7.72 6.16 8.14 6.13 1.08 0 1.9-1.71 1.9-4s-.84-4-1.9-4c-.65 0-1.77.52-1.88 2.55-.07 1.53.63 3.43 1.53 3.54.47.06.89-.76 1-1.6s.06-1.87-.59-2a.38.38 0 0 0-.46.28 3.83 3.83 0 0 1 .45 1.31c0 1.25-1.28.63-1.12-1.36.12-1.74 1.01-1.65 1.05-1.65.5 0 1.26 1 1.26 3 0 1.75-.84 3.63-2.46 2.65-1.36-1-1.89-3.28-1.52-5 .17-.81.87-3 3.13-3 3.26 0 6.3 1.71 8.72 4.9-.27.85-1.95 4.1-7.28 7.21l-.29.15a11 11 0 0 0-4.93-1 27.07 27.07 0 0 0-4.64.74 4.09 4.09 0 0 1-.92.15 4.09 4.09 0 0 1-.92-.15 27.07 27.07 0 0 0-4.64-.74 11 11 0 0 0-4.93 1l-.29-.15c-5.34-3.11-7-6.36-7.28-7.21 2.42-3.19 5.46-4.9 8.72-4.9 2.26 0 3 2.21 3.13 3 .38 1.77-.16 4.05-1.52 5-1.61 1-2.46-.9-2.46-2.65 0-2 .76-3 1.26-3 0 0 .94-.11 1.09 1.65.17 2-1.09 2.61-1.12 1.36a3.83 3.83 0 0 1 .39-1.34.38.38 0 0 0-.43-.28c-.65.16-.71 1.3-.59 2s.56 1.66 1 1.6c.9-.12 1.6-2 1.52-3.44-.1-2-1.23-2.55-1.88-2.55-1.06 0-1.9 1.71-1.9 4s.82 4 1.9 4c.42 0 8.81 1.18 8.14-6.13a9.4 9.4 0 0 0-3-6.1 6.6 6.6 0 0 0-2.38-1.32 13.38 13.38 0 0 0-1.71-.44 4.19 4.19 0 0 1 6.04-1.58ZM10.03 35.82a32.49 32.49 0 0 0-.28 3.31 44.42 44.42 0 0 0 2 15.25 49.48 49.48 0 0 0 4.13 9.32 11.48 11.48 0 0 1-5.13 2.33 66.66 66.66 0 0 1-6.47-31.82 16.13 16.13 0 0 0 5.75 1.61Zm-5.72-2.18c.44-7.31 2.29-13.05 5-16.87.48 1.24 2.57 4.35 7.39 7.18-4.1 2.47-6 7.56-6.58 11.36a15.81 15.81 0 0 1-5.81-1.67Zm11.85 30.53c3.08 5.3 6.12 8.46 8.45 10.13a11.54 11.54 0 0 1-3.54 4.34c-4-2.7-7.4-7-10.07-12.13a11.81 11.81 0 0 0 5.16-2.34Zm8.92 10.47a12.2 12.2 0 0 0 2.67 1.46 12.2 12.2 0 0 0 2.67-1.46 12 12 0 0 0 3.53 4.34 18.69 18.69 0 0 1-3.58 1.78s0-.09 0-.13c-.26-1.32-2-1.59-2.61-1.59s-2.35.27-2.61 1.59v.13a18.69 18.69 0 0 1-3.58-1.78 12 12 0 0 0 3.51-4.34Zm5.81-.32c2.33-1.67 5.37-4.83 8.45-10.13a11.81 11.81 0 0 0 5.16 2.36c-2.67 5.09-6.06 9.41-10.07 12.09a11.54 11.54 0 0 1-3.54-4.34Zm8.72-10.61a49.48 49.48 0 0 0 4.14-9.34 44.1 44.1 0 0 0 1.66-7.32 44.34 44.34 0 0 0 .34-7.92 32.49 32.49 0 0 0-.32-3.31 16.13 16.13 0 0 0 5.75-1.61 66.66 66.66 0 0 1-6.43 31.82 11.48 11.48 0 0 1-5.14-2.34Zm5.78-28.4c-.62-3.8-2.5-8.8-6.58-11.36 4.82-2.83 6.92-5.94 7.39-7.18 2.69 3.82 4.55 9.56 5 16.87a15.81 15.81 0 0 1-5.81 1.65ZM24.75 84.4a23.29 23.29 0 0 1-5.87-2.93 27.5 27.5 0 0 1-3.25-2.62 31.1 31.1 0 0 1-2.35-2.47q-.76-.88-1.46-1.81a47.49 47.49 0 0 1-5.58-9.69 63.9 63.9 0 0 1-3.4-10.24 70.46 70.46 0 0 1-1.79-10.81 64.57 64.57 0 0 1-.07-10.84c.17-1.92.4-3.85.77-5.76a39.32 39.32 0 0 1 1.4-5.59 31 31 0 0 1 2.1-4.86 20.93 20.93 0 0 1 3.15-4.44 16.19 16.19 0 0 1 4-3.1 13.93 13.93 0 0 1 1.93-.87q.51-.18 1-.32a8.82 8.82 0 0 1 1-.26 14 14 0 0 1 2.56-.21 7.58 7.58 0 0 1 3.88 1 8 8 0 0 1 3.34 6c.39 4.52-4.21 5.23-5.11 5.22-.14 0-.21-.13.24-.59a6.53 6.53 0 0 0 1-5.1c-.44-2.07-1.9-3.69-4-3.69a11.16 11.16 0 0 0-8.12 3.89 22.78 22.78 0 0 0-4.37 7.94 42.69 42.69 0 0 0-2.24 11.39 62.43 62.43 0 0 0 .24 10.63 68.71 68.71 0 0 0 1.94 10.59 60.82 60.82 0 0 0 3.53 9.85 43.36 43.36 0 0 0 5.48 9 25.89 25.89 0 0 0 8.13 6.93 18.15 18.15 0 0 0 2.21 1 6.71 6.71 0 0 0 .71 3 6.63 6.63 0 0 1-1-.24Zm3.29 1.55a.6.6 0 0 1-.31.21.6.6 0 0 1-.31-.21 6.49 6.49 0 0 1-1.51-5.17c.12-.64 1.2-.93 1.82-.94s1.7.3 1.82.94a6.49 6.49 0 0 1-1.49 5.16Zm26.41-42.12a70.46 70.46 0 0 1-1.79 10.81 63.9 63.9 0 0 1-3.42 10.2 47.49 47.49 0 0 1-5.58 9.69q-.7.93-1.46 1.81a31.1 31.1 0 0 1-2.35 2.47 27.5 27.5 0 0 1-3.25 2.62 23.29 23.29 0 0 1-5.85 2.97 6.63 6.63 0 0 1-1 .24 6.71 6.71 0 0 0 .67-3.1 18.15 18.15 0 0 0 2.21-1 25.89 25.89 0 0 0 8.13-6.87 43.36 43.36 0 0 0 5.48-9 60.82 60.82 0 0 0 3.53-9.85 68.71 68.71 0 0 0 1.98-10.55 62.43 62.43 0 0 0 .24-10.63 42.69 42.69 0 0 0-2.24-11.39 22.78 22.78 0 0 0-4.47-7.87 11.16 11.16 0 0 0-8.12-3.89c-2.12 0-3.58 1.62-4 3.69a6.53 6.53 0 0 0 1 5.1c.45.46.38.59.24.59-.9 0-5.51-.71-5.11-5.22a8 8 0 0 1 3.34-6 7.58 7.58 0 0 1 3.88-1 14 14 0 0 1 2.56.21 8.77 8.77 0 0 1 1 .26q.52.14 1 .32a13.93 13.93 0 0 1 1.93.87 16.19 16.19 0 0 1 4 3.1 20.93 20.93 0 0 1 3.15 4.44 31 31 0 0 1 2.2 4.79 39.32 39.32 0 0 1 1.4 5.59c.35 1.91.58 3.84.74 5.77a64.57 64.57 0 0 1-.04 10.83Z"/><path fill="#c5050c" d="M44.75 39.27c-.11-2.69-.9-10.9-6.48-14.46a9.41 9.41 0 0 0-4.34-1.17c-2.74-.18-4.77.87-6.14.91h-.04c-1.37 0-3.42-1.09-6.16-.91a9.35 9.35 0 0 0-4.37 1.21c-5.62 3.52-6.4 11.73-6.47 14.42a43.52 43.52 0 0 0 3.54 19c3.82 8.49 8.46 14.51 13.09 16.56l.37.16.37-.16c4.61-2 9.27-8.06 13.14-16.57a43.53 43.53 0 0 0 3.49-18.99Z"/><path fill="#fff" d="M36.05 32.82v1.82h.55a1.17 1.17 0 0 1 .82.28 1.27 1.27 0 0 1 .21 1.11s-3.74 16.19-4.45 19.27l-5.26-25.18v-.09h-.9v.1l-3.87 24.83-4.4-18.95a2.7 2.7 0 0 1 0-.28 1.27 1.27 0 0 1 .31-1 1 1 0 0 1 .69-.17h.37v-1.8h-6.24v1.76h.28a1.16 1.16 0 0 1 .95.83l7.64 32.77v.09h1.1v-.1l3.56-23.3 4.53 23.31v.09h1l7.25-32.78a1 1 0 0 1 1-.75h.43v-1.86h-5.57Z"/></svg>
```

File: entries/w/wisc.edu.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+    "University of Wisconsin Madison": {
+        "domain": "wisc.edu",
+        "url": "https://www.wisc.edu/",
+        "tfa": [
+            "custom-software",
+            "custom-hardware"
```
Comment (Carlgo11) [inline entries/w/wisc.edu.json]: According to https://kb.wisc.edu/helpdesk/90562, U2F keys are also supported.
```suggestion
            "custom-software",
            "custom-hardware",
            "u2f"
```

File: entries/w/wisc.edu.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+        ],
```
Comment (Carlgo11) [inline entries/w/wisc.edu.json:14]: The custom hardware and software offered needs to be specified:
```suggestion
        ],
        "custom-software": [
          "Duo"
        ],
        "custom-hardware": [
          "Duo TOTP key"
        ]
```

**Review by Carlgo11 — CHANGES_REQUESTED** [review]

**Review by Carlgo11 — APPROVED** [review]

File: entries/w/wisc.edu.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{
+    "University of Wisconsin Madison": {
+        "domain": "wisc.edu",
+        "url": "https://www.wisc.edu/",
```
Comment (phallobst) [inline entries/w/wisc.edu.json]:
```suggestion
```

**Review by phallobst — CHANGES_REQUESTED** [review]

**Review by phallobst — APPROVED** [review]

**Pull request merged**